distribution_metadata.c

```diff
@@ -761,12 +764,58 @@ UpdateShardPlacementRowState(int64 shardPlacementId, ShardState newState)
 
 
 /*
- * LockShard returns after acquiring a lock for the specified shard, blocking
- * indefinitely if required. Only the ExclusiveLock and ShareLock modes are
- * supported. Locks acquired with this method are released at transaction end.
+ * LockShardData returns after acquiring a lock for the specified shard's data,
+ * blocking if required. Only ExclusiveLock and ShareLock modes are supported.
+ * Locks acquired with this method are released at transaction end.
  */
 void
-LockShard(int64 shardId, LOCKMODE lockMode)
+LockShardData(int64 shardId, LOCKMODE lockMode)
+{
+	AcquireShardLock(shardId, SHARD_LOCK_DATA, lockMode);
+}
+
+
+/*
+ * LockShardDistributionMetadata returns after grabbing a lock for distribution
+ * metadata related to the specified shard, blocking if required. ExclusiveLock
+ * and ShareLock modes are supported. Any locks acquired using this method are
+ * released at transaction end.
+ */
+void
+LockShardDistributionMetadata(int64 shardId, LOCKMODE lockMode)
+{
+	AcquireShardLock(shardId, SHARD_LOCK_METADATA, lockMode);
+}
+
+
+/*
+ * LockRelationDistributionMetadata returns after getting a the lock used for a
+ * relation's distribution metadata, blocking if required. Only ExclusiveLock
+ * and ShareLock modes are supported. Any locks acquired using this method are
+ * released at transaction end.
+ */
+void
+LockRelationDistributionMetadata(Oid relationId, LOCKMODE lockMode)
+{
+	Assert(lockMode == ExclusiveLock || lockMode == ShareLock);
+
+	if (lockMode == ExclusiveLock)
```

Comment:
Why do you map stuff between ExclusiveLock and AccessExclusiveLock? That seems like a pretty obscure thing to do for me?